pkg/archive/archive.go

```diff
@@ -382,6 +382,56 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 	return nil
 }
 
+// Matches returns true if relFilePath matches any of the patterns
+// and isn't excluded by any of the subsequent patterns.
```

Comment:
@duglin I wasn't going to add docs until I knew whether the function could be used in both contexts (hence porting Exclusion and Empty back into fileutils as they are generic).